### PR TITLE
Pass through $PATH to find dotnet

### PR DESCRIPTION
**Pull request by easwarh**
This commit makes it easier to build on Linux from source. Related to #606, but
applies to x64 builds as well.

Signed-off-by: Easwar Hariharan <easwar.hariharan@microsoft.com>

### CODE REVIEW

**Review by mjcheetham — CHANGES_REQUESTED** [review]
Hello 👋!

Thanks for the contribution. I'm in favour of making it easier to build GCM on more platforms, but I have one concern about this change as it stands.

The `$DOTNET_ROOT` variable is set by the `dotnet` app host executable, which is typically the caller of these scripts (via MSBuild). However in the cases that this script is called directly (not via `dotnet`) then the variable is not set, and the commands being run in this script will be `/dotnet publish ...` which will not resolve correctly.

If we are to accept this then there should be a check that if the `$DOTNET_ROOT` variable is not set, then the script should fall back to invoking `dotnet` and let the shell resolve the executable location.. 
..OR add something like this to explicitly resolve:

```shell
if [ -z "$DOTNET_ROOT" ]; then
    DOTNET_ROOT="$(dirname $(which dotnet))"
fi
```

**Comment by easwarh** [comment]
Hi @mjcheetham, that's a great call out, thanks for the feedback! I'll fix this up, would you rather have a fixup commit (commit --fixup), a new commit, or an amendment to the existing commit (commit --amend)?

**Comment by ldennington** [comment]
@easwarh - I know you directed your question at Matthew, but FWIW my usual workflow in GCM is to amend my existing commits.

**Comment by easwarh** [comment]
> @easwarh - I know you directed your question at Matthew, but FWIW my usual workflow in GCM is to amend my existing commits.

Thanks for the headsup. I'll push a merged commit.

**Review by mjcheetham — APPROVED** [review]
Looks good to me!

File: src/linux/Packaging.Linux/build.sh [hunk 1]
```diff
@@ -106,9 +106,13 @@ else
...
 fi
 
+if [ -z "$DOTNET_ROOT" ]; then
+    DOTNET_ROOT="$(dirname $(which dotnet))"
+fi
+
 # Publish core application executables
 echo "Publishing core application..."
-dotnet publish "$GCM_SRC" \
+$DOTNET_ROOT/dotnet publish "$GCM_SRC" \
```
Comment (Osama2535) [inline src/linux/Packaging.Linux/build.sh:115]: Oatef4313@gmail.com

**Pull request merged**